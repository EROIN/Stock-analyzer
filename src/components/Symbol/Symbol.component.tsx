import './Symbol.scss';

import {SymbolProps} from './Symbol.types';

export const Symbol = (props: SymbolProps) => {
  const {data} = props;
  return (
    <>
      <div className="symbol-container">
        {data['2. name']} ({data['1. symbol']})
      </div>
      <div className="symbol-details">
        Open: {data['5. marketOpen']}, Close: {data['6. marketClose']}
      </div>
    </>
  );
};

// added so that if someone doesnt pass by mistake, it doesnt break the page
Symbol.defaultProps = {
  data: {},
};
