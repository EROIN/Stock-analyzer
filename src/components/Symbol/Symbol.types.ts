import {SymbolSearch} from '../../types/search/symbolSearch.types';

export interface SymbolProps {
  data: SymbolSearch;
}
