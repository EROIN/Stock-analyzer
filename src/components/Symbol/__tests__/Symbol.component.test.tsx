import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import {Symbol} from '../Symbol.component';
// eslint-disable-next-line jest/no-mocks-import
import {SYMBOLS} from '../../../__mocks__/symbolSearch.mocks';

describe('<Symbol/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(<Symbol data={SYMBOLS[0]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with empty data', () => {
    const tree = renderer
      //@ts-ignore
      .create(<Symbol data={{}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
