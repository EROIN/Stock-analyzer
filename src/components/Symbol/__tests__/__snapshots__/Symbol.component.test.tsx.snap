// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Symbol/> renders correctly with default props 1`] = `
Array [
  <div
    className="symbol-container"
  >
    IsoEnergy Ltd
     (
    I01.FRK
    )
  </div>,
  <div
    className="symbol-details"
  >
    Open: 
    08:00
    , Close: 
    20:00
  </div>,
]
`;

exports[`<Symbol/> renders correctly with empty data 1`] = `
Array [
  <div
    className="symbol-container"
  >
     (
    )
  </div>,
  <div
    className="symbol-details"
  >
    Open: 
    , Close: 
  </div>,
]
`;
